treeherder/webapp/api/performance_data.py

```diff
@@ -423,3 +426,63 @@ class PerformanceIssueTrackerViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = IssueTrackerSerializer
     filter_backends = [filters.OrderingFilter]
     ordering = 'id'
+
+
+class PerformanceByRevision(generics.ListAPIView):
+
+    serializer_class = PerformanceRevisionSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = PerformanceQueryParamsSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors,
+                            status=HTTP_400_BAD_REQUEST)
+
+        startday = query_params.validated_data['startday']
+        endday = query_params.validated_data['endday']
+        revision = query_params.validated_data['revision']
+        repository = query_params.validated_data['repository']
+        interval = query_params.validated_data['interval']
+        frameworks = query_params.validated_data['framework']
+        parent_signature = query_params.validated_data['parent_signature']
+        no_subtests = query_params.validated_data['no_subtests']
+
+        signature_data = (PerformanceSignature.objects
+                                              .select_related('framework', 'repository', 'platform', 'push')
+                                              .filter(repository__name=repository, framework__in=frameworks,
+                                                      parent_signature__isnull=no_subtests))
+        if parent_signature:
+            signature_data = signature_data.filter(parent_signature__signature_hash=parent_signature)
+
+        if interval:
+            signature_data = signature_data.filter(last_updated__gte=datetime.datetime.utcfromtimestamp(
+                                                   int(time.time() - int(interval))))
+
+        signature_ids = signature_data.values_list('id', flat=True)
+
+        self.queryset = (signature_data.values('framework_id', 'id', 'lower_is_better', 'has_subtests', 'extra_options', 'suite',
+                                               'signature_hash', 'platform__platform', 'test', 'option_collection__option__name',
+                                               'parent_signature__signature_hash'))
+
+        data = (PerformanceDatum.objects.select_related('push', 'repository')
+                                .filter(signature_id__in=signature_ids, repository__name=repository))
+
+        if revision:
+            data = data.filter(push__revision=revision)
```

Comment:
Since we're here anyway, I think we should redesign this method to return data for both revisions at once. That should speed things up further.

Reply:
I'm already seeing a 503 for the query with the date range on prototype that I didn't see while testing locally with the prod db mirror so I'll address that first and then see about this. The two queries are being called asynchronously in the UI so I think that will help with performance.

Reply:
Generally I think it's better to get all the data you can in one query, if it's possible/reasonable.

Reply:
Fair enough. Two API calls isn't the end of the world.